sklearn/base.py

```diff
@@ -398,6 +416,54 @@ def __sklearn_tags__(self):
             classifier_tags=None,
         )
 
+    # TODO(1.7): Remove this method
+    def _more_tags(self):
+        warnings.warn(
+            "The `_more_tags` method is deprecated in 1.6 and will be removed in "
+            "1.7. Please implement the `__sklearn_tags__` method.",
+            category=FutureWarning,
+        )
+        return _to_old_tags(default_tags(self))
+
+    # TODO(1.7): Remove this method
+    def _get_tags(self):
+        warnings.warn(
+            "The `_get_tags` tag provider is deprecated in 1.6 and will be removed in "
+            "1.7. Please implement the `__sklearn_tags__` method.",
+            category=FutureWarning,
+        )
+        # In case a user called `_get_tags` but that the estimator already did the job
+        # implementing `__sklearn_tags__` completely and removed `_more_tags`, let's
+        # default back to the future behaviour. Otherwise, we will get the default tags.
+        from sklearn.utils._tags import _find_tags_provider, _to_old_tags, get_tags
+
+        if _find_tags_provider(self, warn=False) == "__sklearn_tags__":
+            return _to_old_tags(get_tags(self))
+
+        collected_tags = {}
+        for base_class in reversed(inspect.getmro(self.__class__)):
+            if hasattr(base_class, "_more_tags"):
+                # need the if because mixins might not have _more_tags
+                # but might do redundant work in estimators
+                # (i.e. calling more tags on BaseEstimator multiple times)
+                with warnings.catch_warnings():
+                    warnings.filterwarnings(
+                        "ignore",
+                        category=FutureWarning,
+                        message="The `_more_tags` method is deprecated",
+                    )
+                    # silence the warning to avoid a false positive warning since this
+                    # is not a direct user call but an internal one.
+                    more_tags = base_class._more_tags(self)
+                collected_tags.update(more_tags)
+            elif hasattr(base_class, "__sklearn_tags__"):
+                # Since that some people will inherit from scikit-learn that implements
+                # the new infrastructure, we need to collect it and merge it with
+                # the old tags.
+                more_tags = base_class.__sklearn_tags__(self)
+                collected_tags.update(_to_old_tags(more_tags))
```

Comment:
Can you provide me a case where it would break?

Reply:
So this would be:

```python
class Mixin:
    def __sklearn_tags__(self):
        tags = super().__sklearn_tags__()
        tags.input_tags.allow_nan = True
        return tags

class MyClass(Mixin, BaseEstimator):

    def _more_tags(self):
        return {
            "multioutput": True
        }
```

and indeed it end up with a recursion error.